artemis-core-client/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/NettyConnection.java

```diff
@@ -395,19 +425,18 @@ private static void flushAndWait(final Channel channel, final ChannelPromise pro
       }
    }
 
-   private ChannelFuture writeBatch(final ByteBuf bytes, final int readableBytes, final ChannelPromise promise) {
+   private ChannelFuture writeBatch(final ByteBuf bytes, Object fileObject, final int readableBytes, final ChannelPromise promise) {
       final int batchBufferSize = batchBufferSize(channel, this.writeBufferHighWaterMark);
       final int nextBatchSize = batchBufferSize + readableBytes;
       if (nextBatchSize > batchLimit) {
          //request to flush before writing, to create the chance to make the channel writable again
          channel.flush();
-         //let netty use its write batching ability
-         return channel.write(bytes, promise);
-      } else if (nextBatchSize == batchLimit) {
-         return channel.writeAndFlush(bytes, promise);
+      }
+      if (fileObject == null) {
+         return nextBatchSize == batchLimit ? channel.writeAndFlush(bytes, promise) : channel.write(bytes, promise);
       } else {
-         //let netty use its write batching ability
-         return channel.write(bytes, promise);
+         channel.write(bytes);
+         return nextBatchSize == batchLimit ? channel.writeAndFlush(fileObject, promise) : channel.write(fileObject, promise);
```

Comment:
Try to check if it makes sense to batch writes if we are using ChunkedFile/DefaultFileRegion